### PR TITLE
Update handlebars package due to a Security Vulnerability

### PR DESCRIPTION
**Pull request by adelriosantiago**
Update handlebars package due to a Security Vulnerability

### CODE REVIEW

**Comment by adelriosantiago** [comment]
**Explanation**

The handlebars package is vulnerable to Prototype Pollution resulting from Improper Input Validation. Multiple functions in multiple files, as listed below, do not disallow the calling of helperMissing and blockHelperMissing helpers directly. This can enable an attacker to conduct remote code execution attacks by calling methods from object prototypes which should otherwise not have been called.

**Vulnerable File(s) and Function(s)**

package/lib/handlebars/compiler/javascript-compiler.js

package/dist/amd/handlebars/compiler/javascript-compiler.js

package/dist/cjs/handlebars/compiler/javascript-compiler.js

package/lib/handlebars/helpers/runtime.js

invokePartialWrapper()
package/dist/cjs/handlebars/helpers/runtime.js

invokePartialWrapper()
package/dist/amd/handlebars/helpers/runtime.js

invokePartialWrapper()
package/dist/handlebars.runtime.js

invokePartialWrapper()
package/dist/handlebars.runtime.amd.js

invokePartialWrapper()
package/dist/handlebars.js

invokePartialWrapper()
package/dist/handlebars.amd.js

invokePartialWrapper()
package/dist/handlebars.runtime.min.js

package/dist/handlebars.runtime.amd.min.js

package/dist/handlebars.min.js

package/dist/handlebars.amd.min.js

**Comment by UziTech** [comment]
@sahat could you merge this PR to remove the security vulnerability?

**Comment by pgnedoy** [comment]
@sahat please merge this PR

**Review by pgnedoy — APPROVED** [review]

**Review by milo526 — APPROVED** [review]

**Comment by adelriosantiago** [comment]
Any update on this? We currently rely on this package and we are unable to pass the security checklist.

@sahat

**Review by Jupp56 — APPROVED** [review]
Should fix the issue without side effects; Changes on the handlebars side are non-breaking and tested.

**Comment by milo526** [comment]
@adelriosantiago Could you please update this MR to reflect the new advisories?

https://www.npmjs.com/advisories/1300
https://www.npmjs.com/advisories/1316
https://www.npmjs.com/advisories/1324
https://www.npmjs.com/advisories/1325

**Remediation**
Upgrade to version 4.5.3 or later.

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   "dependencies": {
     "glob": "^7.1.3",
     "graceful-fs": "^4.1.2",
-    "handlebars": "^4.1.2",
+    "handlebars": "^4.5.1",
```
Comment (milo526) [inline package.json]: Handlebars ^4.5.1 could still be vulnerable to Arbitrary Code Execution and Prototype Pollution. Please upgrade this to ```^4.5.3```

**Review by milo526 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (adelriosantiago) [reply package.json]: Fixed, updated the PR, thanks for the observation.

**Review by milo526 — APPROVED** [review]

**Comment by UziTech** [comment]
I have finally gotten around to moving this repo to https://github.com/express-handlebars/express-handlebars/ any new development will be done on that repo.

handlebars is up to date in v4.0.0 so this can be closed.

**Comment by adelriosantiago** [comment]
Ok, closing this. Thanks!

**Pull request closed without merging**